conda_libmamba_solver/solver.py

```diff
@@ -239,6 +264,68 @@ def __init__(
             fixed_specs.append(spec)
         self.specs_to_add = frozenset(MatchSpec.merge(s for s in fixed_specs))
 
+    def _notify_conda_outdated(
+        self,
+        link_precs,
+        index: LibMambaIndexHelper = None,
+        final_state: Iterable[PackageRecord] = None,
+    ):
+        """
+        We are overriding the base class implementation, which gets called in
+        Solver.solve_for_diff() once 'link_precs' is available. However, we
+        are going to call it before (in .solve_final_state(), right after the solve).
+        That way we can reuse the SolverInputState instance we have lying around,
+        which contains the index information we need, before we lose it.
+        """
+        if index is None and final_state is None:
+            return
+        if not context.notify_outdated_conda or context.quiet:
+            return
+        current_conda_prefix_rec = PrefixData(context.conda_prefix).get("conda", None)
+        if not current_conda_prefix_rec:
+            return
+
+        channel_name = current_conda_prefix_rec.channel.canonical_name
+        if channel_name == UNKNOWN_CHANNEL:
+            channel_name = "defaults"
+        # only look for a newer conda in the channel conda is currently installed from
+        conda_newer_str = f"{channel_name}::conda>4.10"
+        conda_newer_spec = MatchSpec(conda_newer_str)
+
+        # if target prefix is the same conda is running from
+        # maybe the solution we are proposing already contains
+        # an updated conda! in that case, we don't need to check further
+        if paths_equal(self.prefix, context.conda_prefix):
+            if any(conda_newer_spec.match(record) for record in final_state):
+                return
+
+        # check if the index contains records that match a more recent conda version
+        conda_newer_records = index.search(conda_newer_str)
+
+        # print instructions to stderr if we found a newer conda
+        if conda_newer_records:
+            newest = max(conda_newer_records, key=lambda x: VersionOrder(x.version))
+            latest_version = newest.version
+            # If conda comes from defaults, ensure we're giving instructions to users
+            # that should resolve release timing issues between defaults and conda-forge.
+            add_channel = "-c defaults " if channel_name == "defaults" else ""
+            print(
+                dedent(
+                    f"""
+
+                    ==> WARNING: A newer version of conda exists. <==
+                        current version: {_conda_version}
+                        latest version: {latest_version}
+
+                    Please update conda by running
+
+                        $ conda update -n base {add_channel}conda
+
+                    """
+                ),
+                file=sys.stderr,
+            )
+
```

Comment:
With the new release policies and improved release times, I wonder if it wouldn't be useful to also print the channel name for non-defaults channels here?

```suggestion
            print(
                dedent(
                    f"""

                    ==> WARNING: A newer version of conda exists. <==
                        current version: {_conda_version}
                        latest version: {latest_version}

                    Please update conda by running

                        $ conda update -n base -c {channel_name} conda

                    """
                ),
                file=sys.stderr,
            )

```

Reply:
Nice! Btw, this will need to be updated in [`conda`'s implementation too](https://github.com/conda/conda/blob/e69b9efd5fc96e26bb2aa3b180e3c504c65ce65f/conda/core/solve.py#L1019-L1029), @jezdez